dask/array/core.py

```diff
@@ -1367,7 +1391,11 @@ def handle_nonmatching_names(func, args, kwargs):
 
             return _HANDLED_FUNCTIONS[func](*args, **kwargs)
 
-        # First try to find a matching function name.  If that doesn't work, we may
+        # First, verify that all types are handled by Dask. Otherwise, return NotImplemented.
+        if not all(type is Array or is_valid_chunk_type(type) for type in types):
```

Comment:
@jthielen what was your reasoning for `type is Array` versus `issubclass(type, Array)`?

This breaks subclassing `da.Array`—or at least `__array_function__` dispatch no longer works on subclasses. I don't want to `register_chunk_type` on my `da.Array` subclass, since it's _not_ actually a valid chunk type.

I'm happy to open a PR switching to `issubclass` if you think that'll work, but I haven't read this whole PR carefully enough to follow all the nuances.